ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -9607,6 +9610,41 @@ void HqlCppTranslator::doBuildExprFailCode(BuildCtx & ctx, IHqlExpression * expr
     }
 }
 
+void HqlCppTranslator::doBuildExprMatchedInJoin(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    HqlExprAssociation * matchedMarker = ctx.queryMatchExpr(activeJoinTransformFlagsMarkerExpr);
+    if (!matchedMarker)
+    {
+        if (!buildExprInCorrectContext(ctx, expr, tgt, false))
+            throwError(HQLERR_MatchedContextNotFound);
+        return;
+    }
+
+    __int64 mask = 0;
```

Comment:
@ghalliday I think if you could help create a test case that would help more than anything.  There is something about matching the correct layer of nested joins right and left that is probably quite straight forward but that I'm just not yet picturing fully.